packages/dds/tree/src/feature-libraries/optional-field/optionalField.ts

```diff
@@ -178,7 +178,7 @@ export const optionalChangeRebaser: FieldChangeRebaser<OptionalChangeset> = {
 			return { srcToDst, dstToSrc };
 		};
 
-		let latestReservedDetachId: RegisterId | undefined;
+		let earliestReservedDetachId: RegisterId | undefined;
```

Comment:
Some axiomatic tests suggested that this might be better, though the choice is somewhat arbitrary (there were things not quite right with the test setup which convinced me to hold off checking them in to a later date, but this change seemed harmless enough to make now)